compiler/luci-interpreter/include/luci_interpreter/Interpreter.h

```diff
@@ -64,7 +65,11 @@ class Interpreter
   const Tensor *getTensor(const loco::Node *node) { return _node_to_tensor[node]; }
 
 private:
+  //_default_memory_manager should be before _runtime_module due to
+  // the order of deletion in the destructor
+  std::unique_ptr<IMemoryManager> _default_memory_manager = nullptr;
```

Comment:
(optional) I think it would be nice to make `_memory_manager` as `unique_ptr` and remove `_default_memory_manager` member, but this is requires a lot of changes in this and next PR's, so, I'm okay with current design =)

Reply:
The main idea why we have two memory_managers is that we want that some tools (like EvalDriver) could have own memory manager and interpreter can use it. But if tool hasn't memory manager interpreter creates default_memory_manager and uses it.

Reply:
Let me add some details:
We want to have several variants of interpreter interfaces:
- Old simple interface, when you create interpreter and good to go
- New customizable interface which can be used if you want to adjust behavior for specific use case

To reduce code duplication while adding new options (like custom memory manager in this PR) we can implement customizable implementation and define default values for options.

Problem with memory manager is following:
- If simplified interface is used, we need to create memory manager inside interpreter, so luci-interpreter owns memory manager and should delete it.
- If full interface is used, memory manager is created outside of interpreter and could be shared between multiple interpreters. luci-interpreter does not own this memory manager and should not delete it.

`_default_memory_manager` is needed in case we use simplified interface and need to remove MM with luci-interpreter.

Reply:
Ah, I got it! =)

But I imagined following desing:
1. If we use `Interpreter` and would like to provide own memory manager, we need to create manger wrapped to `unique_ptr` like:
```cpp
auto manager = std::make_unique<CustomManager>();
auto const *manager_ptr = manager.get(); // if we need call some const managers methods later
```
2. Then,  **move** unique_ptr to Interpreter's constructor (give ownership to manager). 
```cpp
auto interpreter = std::make_unique<Interpreter>(module, std::move(manager));
```
3. Constructor of Interpreter expects rvalue, `_memory_manager` is unique and manger set up will be something like:
```cpp
  if (memory_manager == nullptr)
  {
    _memory_manager = std::make_unique<SimpleMemoryManager>();
  }
  else
  {
    _memory_manager = std::move(memory_manager);
  }
```

Reply:
But this is a not requirement, just a note. Current design is good to me too!

> Ah, I got it! =)
> 
> But I imagined following desing: